### DISCUSSION TITLE
Progress List

### DISCUSSION THREAD

**Discussion by Rozerinay** [nl-design-system/backlog#250] [Component suggestions]
<img width="480" alt="visuele weergave van het progress list component" src="https://github.com/nl-design-system/backlog/assets/84851545/7cc93ea2-f4c2-4a31-8e6e-bc2c03ebbb87">

We willen graag meer leren over het Progress List component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1715761405493259)

---

## Naam
Progress List

## Doel
Toont de voortgang van een reeks stappen en biedt indien nodig extra details over een stap.

## Ook bekend als
Progress, Progress Indicator

## Gerelateerde componenten

[Form Navigation](https://github.com/orgs/nl-design-system/discussions/355)

## Onderzoek
[Onderzoek Status Component van MijnZaken - Gebruikersonderzoeken](https://gebruikersonderzoeken.nl/docs/onderzoek-bekijken/mijn-zaken/utrecht-status-component)
Onderzoek naar het 'Status' component door gemeente Den Haag en Utrecht. 

[Onderzoek Generieke Services - Miro](https://miro.com/app/board/uXjVP0OE2fg=/?share_link_id=511269957959)
Miro board dat is gebruikt tijdens het gebruikersonderzoek waarin inzichten te vinden zijn over de Progress List. 

[Contactformulier voor WMEBV - Gebruikersonderzoeken](https://gebruikersonderzoeken.nl/docs/onderzoek-bekijken/wmebv/vng-online-formulieren)
Onderzoek waarin een inzicht te vinden is over de Progress List. 

[Status Trackers and Progress Updates: 16 Design Guidelines - NN Group](https://www.nngroup.com/articles/status-tracker-progress-update/)
Artikel waarin ontwerprichtlijnen worden gedeeld voor de status tracker. 

[Nieuw ontwerp Zakelijk erfpachtportaal - gemeente Amsterdam](https://codefornl.slack.com/archives/C7KDQRXMZ/p1707223988909559)
In dit onderzoek is te lezen dat de Progress List handig wordt gevonden, mits er een indicatie te zien is van wanneer de volgende stap wordt bereikt.

**Comment by Rozerinay** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-processsteps--docs)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/ProcessSteps)
- [Figma - Huidig](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/🌻-Den-Haag-Design-System?type=design&node-id=16469-107354&mode=design&t=u2n2umaqcb4GoEPc-11)

---

### Screenshots / Varianten

**Storybook**

Default
<img width="417" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/95a4db6c-9283-42ac-a6cf-60158ac6fe3a">

Collapsible
<img width="387" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/d31a858d-032a-419f-b4f7-12a1ef95f69e">

Collapsible with expanded step
<img width="375" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/efcc13f7-63a2-4181-ad46-aabdbae89ba3">

Without substeps
<img width="638" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/63a6a805-9a52-4314-87f9-7be4b7a165da">

With metadata
<img width="659" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/7b95dd8b-eff6-4fe5-b32d-1fcd7f758b26">

With expanded metadata
<img width="938" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/215b4cf9-11dd-4175-805f-c86574096560">

With warning
<img width="414" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/656d0227-f1f0-4538-99ff-17a7231345d7">

With error
<img width="951" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/68bf5160-767a-49c1-9d94-097c62e831cc">

Non collapsible step
<img width="951" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/ce5c5ee8-7d74-481f-ac75-7cb8720d6971">

**Figma - Huidig**
Large viewport
<img width="394" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/ae15dd72-02e4-4709-9849-8d834167ee2a">

Small viewport
<img width="396" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/8fd7ba3f-1f62-4224-9ae3-7fb54e514a93">

---

### Notities
- Onder de naam: Process steps.
- Icoon: arrow down (arrow up bij active)
- Richtlijnen in Figma beschikbaar.
- In Figma is een sub step date te vinden. Deze komt niet voor in Storybook.
- In Figma is een disabled state te vinden dat is afgeleid van de Progress indicator. Deze komt niet voor in Storybook.
- In Storybook is de step meta ook (deels) beschikbaar als link. In Figma niet.
- De nested step markers verschillen in Storybook en Figma.

**Comment by Rozerinay** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/timeline)

---

### Screenshots / Varianten

Basic
<img width="458" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/73b6d3f0-e5a2-44b2-92de-b7ad857df559">

Compacte tijdlijn
<img width="226" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/d784d300-c03e-466a-afbc-0001db780661">

<img width="919" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/42d14241-3b97-4098-8cc4-e03f36f0b77a">


Grote tijdlijn
<img width="774" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/7bbbd9a3-833a-4af2-a6ea-da1b427fa402">

---

### Notities
- Onder de naam: Timeline.
- De stappen lopen in chronologische volgorde (van oud naar nieuw).
- De variant 'Compacte tijdlijn' komt het meest overeen met het Progress list component, omdat de de voortgang van een reeks stappen wordt getoond. 
- De variant 'Compacte tijdlijn' is niet interactief.
- Het ontwerp van de variant 'Basic tijdlijn' verschilt in Vanilla JS en Angular.
- Er wordt gebruik gemaakt van een badge om de status weer te geven.
- Documentatie beschikbaar per variant.

**Comment by Rozerinay** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/react_react-process-steps--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/process-steps)

---

### Screenshots / Varianten

**Storybook**

Default
<img width="353" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/d93014a5-2b52-4508-acfc-1c6da2db5ad4">


---

### Notities
- Onder de naam: Process steps.
- Is gebaseerd op de white-label component die bij Gemeente Den Haag is ontwikkeld.

**Comment by jiromaykin** [comment]
In OpenInwoner hebben wij een accordeon-like component voor alle verschillende 'statuses' van een aanvraag.
De laatste stap heeft een andere kleur; afhankelijk van de soort status (afgehandeld/in behandeling/actie vereist/etc/etc.
Aanvragen die nog in behandeling zijn, hebben een iets andere extra vorm; die tonen een 'toekomstige' status zonder uitklapbaar status bericht.

PY code hier: https://github.com/maykinmedia/open-inwoner/blob/develop/src/open_inwoner/templates/pages/cases/statuses.html 

<img width="791" alt="OIP-aanvragen-statuses" src="https://github.com/nl-design-system/backlog/assets/118177951/2b0ffe98-2011-4ee2-b1fb-ff612647a5c9">

**Reply by jiromaykin** [reply]
Aanvulling: deze zaakstatustypen hebben allen een individueel configureerbare kleur voor de 'huidige' actieve status, 
en, afhankelijk van de back-end/beschikbare status 'trees' is er een toekomstige eind-status te zien.

**Comment by Rozerinay** [comment]
## Amsterdam

Geen specifiek component gevonden in Figma of Storybook. Nico Druif van gemeente Amsterdam geeft dat de [Progress list in het Erfpacht ontwerp](https://www.figma.com/design/jvKCJhmZGKv0fVJxSbxZ8v/🏠-Erfpacht---MijnDenHaag-%2F-Amsterdam?node-id=22915-119331&t=C4aTnn67HQCXDrst-11) staat.

---

### Screenshots / Varianten

**Figma**

<img width="654" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/2ecf6677-4507-4592-b707-b2ca9bc1af3a">

Active
<img width="557" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/c3ee385f-7b78-4ec4-8efb-09d099b47e49">

Done
<img width="565" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/e6f426ed-ecf8-4ffa-af69-0a99d977f059">

Future
<img width="506" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/fd8fea9d-e93c-4e07-9ed6-ca1d14542d14">

---

### Notities

- Onder de naam: Status.
- Bestaat alleen in design.
- De stappen lopen in chronologische volgorde (van oud naar nieuw).
- Bevat op de plek van de datum ook een link.

**Reply by Robbert** [reply]
Amsterdam heeft ook zo'n component in de Storybook van de Amsterdam App, die @VincentSmedinga deelde:

Links:
- [Storybook: Timeline component](https://amsterdam.github.io/amsterdam-app-storybook/?path=/docs/components-ui-containers-timeline--docs)

default:

<img width="620" alt="Screenshot 2024-05-28 at 08 18 27" src="https://github.com/nl-design-system/backlog/assets/30694/0ff980e3-46df-4e54-a989-f120bb57a2ca">

multiple expanded:

<img width="589" alt="Screenshot 2024-05-28 at 08 16 04" src="https://github.com/nl-design-system/backlog/assets/30694/b2b8e594-46e0-42fb-b02d-95e8d596a5a8">

not expanded:

<img width="589" alt="Screenshot 2024-05-28 at 08 15 54" src="https://github.com/nl-design-system/backlog/assets/30694/3812d975-6fac-42b1-89dd-372299dd948e">

**Comment by Rozerinay** [comment]
## Logius

[Rene Olling](https://codefornl.slack.com/archives/C7KDQRXMZ/p1715930376672539?thread_ts=1715761405.493259&cid=C7KDQRXMZ) geeft aan een uitwerking van het [component te zien is in het MijnZaken prototype](https://www.figma.com/design/7FWSxhH1CFbnTeZoiYPeCZ/MijnZaken---MijnOverheid---UX-verkenning?node-id=517-6718&t=E0GXubqlTvFdNCBj-11) waar Anton Olivier van Logius aan werkt. 

---

### Screenshots / Varianten

**Figma**

<img width="700" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/5fd463e4-12e9-403b-be55-cb897c83f934">

<img width="700" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/2b5474f9-0915-43a8-8e12-cc3789546fb9">


---

### Notities
- Bestaat alleen in design.
- De stappen lopen in chronologische volgorde (van oud naar nieuw).
- Hoofdstappen kunnen in- en uitklapbaar zijn.
- Bevat extra details over een stap. 
- Kan substappen bevatten.
- Er wordt gebruik gemaakt van een 'Lees meer…' link in een substap.  
- Kan buttons bevatten.
- Heeft datums van wanneer een stap is bereikt en een toekomstige datum.

**Comment by Rozerinay** [comment]
## OpenInwoner

[Figma](https://www.figma.com/design/iKGhWhstaLIlFSaND2q7cE/OIP---Designs-(new)?node-id=184-6289&t=XOX4FGYhpg7hTkxi-0)

---

### Screenshots / Varianten

**Figma**

<img width="430" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/d0a0c444-17b1-4df1-a8a4-174c4f985de4">

Active
<img width="430" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/d2eeff42-f165-4e66-a514-121a8624a30b">

Caution
<img width="430" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/828a9e05-c059-446f-b198-7bfea3106f53">

Error
<img width="430" alt="image" src="https://github.com/nl-design-system/backlog/assets/84851545/144e4d19-adcf-4a50-8b45-26fc4a36c164">


---

### Notities
- In Figma onder de naam: Progress indicator.
- In code: Statuses.
- Bestaat in design en in Python. 
- De stappen lopen in chronologische volgorde (van oud naar nieuw).
- Hoofdstappen kunnen in- en uitklapbaar zijn.
- Extra details over een stap zijn te zien bij het uitklappen. 
- Kan buttons bevatten.
- Heeft datums van wanneer een stap is bereikt.

**Reply by jiromaykin** [reply]
> ## OpenInwoner
> * Figma https://www.figma.com/design/R4KuAAWc284izFHbushcbH/OIP-Design-System?node-id=1076-12005&t=JbeszloWumZpeRNJ-11)

@Rozerinay Dat is niet het juiste Figma project voor OIP - die is erg verouderd. 
Dit is het correcte Open-Inwoner Figma project met de progress-list achtige statuslijst: https://www.figma.com/design/iKGhWhstaLIlFSaND2q7cE/OIP---Designs-(new)?node-id=184-6289&t=XOX4FGYhpg7hTkxi-0

**Reply by UXTeamGroningen** [reply]
https://gebruikersonderzoeken.nl/docs/onderzoek-bekijken/mijn-zaken/groningen-mijn-omgeving/
Dit design kwam voort uit onderzoek. Hiervoor (tijd geleden) zag de pagina er compleet anders uit.

In dit huidige design is de ervaring dat:
- Inwoners begrijpen de status beter
- De inwoner wil inzicht in het hele proces, daarbij helpt het inzicht in vorige en toekomstige statussen. 
- De substatusfunctie draagt ook daaraan bij. Soms helpt namelijk enkel de naam van de status niet. De substatus geeft meer context over de status en legt deze uit. Dat kan per zaak anders worden ingericht. In het geval van bepaalde producten is soms ook een derde partij betrokken, zoals het RDW of het Rijk, is het fijn te benoemen dat bijvoorbeeld het RDW op dat moment bezig is met de keuring. 
- De knop met vervolgactie 'document uploaden' door naar beneden te scrollen wordt begrepen en gebruikt.

**Reply by jiromaykin** [reply]
Zie ook hoger in deze discussie hier: https://github.com/orgs/nl-design-system/discussions/250#discussioncomment-9454557
Aanvulling: 

- Zaken hebben geen sub-statussen, alleen statussen en metadata + (in Open Inwoner) extra functionaliteit. 
- De scroll-knop kan ook geconfigureerd worden als knop die verwijst naar een (extern) URL waar gebruikers een document kunnen uploaden.

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Progress list op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=progress+list)

**Comment by Robbert** [comment]
## Hoogheemraadschap De Stichtse Rijnlanden

In het wild: [Renovatie gemaal en stuw Groenekan - Hoogheemraadschap De Stichtse Rijnlanden](https://www.hdsr.nl/buurt/bilt/renovatie-gemaal/#PagCls_976846)

<img width="853" alt="Screenshot 2025-03-02 at 23 41 37" src="https://github.com/user-attachments/assets/176a5b93-876b-419b-b0a2-4854999436d4" />

Notities:
- In de CSS heet het "tijdlijn".
- Het toeganklijkheidsrapport merkt aan de de voltooiingsstatus met alleen kleur is aangegeven, en dat dit niet voldoende is.

**Comment by Rozerinay** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-containers-progress-list--docs)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/progress-list)
- [Figma](https://www.figma.com/design/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?node-id=2927-29221&p=f&t=R1MPM62r99ODA12n-11)

---

### Screenshots / Varianten

**Storybook**

Default
<img width="779" height="905" alt="image" src="https://github.com/user-attachments/assets/370de0cb-8eef-47d9-997d-eb86c02c2444" />

With substeps
<img width="781" height="1009" alt="image" src="https://github.com/user-attachments/assets/d973b52e-ae8e-44e2-9edd-1cc5a9ba4f32" />

---

### Notities

- Heading Level kan worden ingesteld op: 2, 3 of 4.
- Stap status kan zijn: Completed, Current of Upcoming.
- Wanneer geen status is ingesteld, gebruikt de stap de default weergave (Upcoming).
- Marker (indicator) per status verschilt.
- Marker kan een icoon bevatten.
- Stap kan qua content bestaan uit een Paragraph en een (Unordered) List.
- Stap is opgebouwd uit: `__indicator` en `__content`.
- Wanneer een stap substappen heeft, bevinden deze zich binnen `__content`.
- Progress List kan alleen worden gebruikt op een lichte achtergrond.
- Op Small Containers verandert de layout van de component niet.
- Documentatie in Storybook beschikbaar.

**Reply by VincentSmedinga** [reply]
> Stap kan qua content bestaan uit een Paragraph en een (Unordered) List.

Een Step toont wat je er in zet, dus in theorie kan dat alles zijn. In de praktijk zullen het inderdaad vaak paragrafen en lijsten zijn.

**Reply by VincentSmedinga** [reply]
Intussen kun je alle stappen van een Progress List ook in- en uitklappen.

<img width="993" height="591" alt="Ams Progress List collapsible" src="https://github.com/user-attachments/assets/7870b3a2-06a7-41c3-98c4-0e872433f036" />

**Comment by jiromaykin** [comment]
## 3 Progress List / Process Steps / Zaakstatussen

Om ervoor te zorgen dat er een goed component komt dat voor iedereen bruikbaar, onderzocht is, voldoet aan de toegankelijkheidseisen, én alle functionaliteit behoudt voor organisaties die deze al op productie hebben staan, hier een overzicht van de verschillen:

### Designs

Den Haag
https://nl-design-system.github.io/denhaag/?path=/story/react-processsteps--collapsible
Amsterdam
https://designsystem.amsterdam/?path=/story/components-containers-progress-list--collapsible
Open Inwoner
https://www.figma.com/design/iKGhWhstaLIlFSaND2q7cE/OIP---Designs--new-?node-id=9269-29387&t=2nGcICTeuiM37Oh4-1

### Verschil in functionaliteit/logica/UX

#### Den Haag mist nog:
- uitklapgedrag op basis van metadata (is nu alleen collapsible op basis van substeps, dus meta-data kan niet worden in- of uit-geklapt)
- gedeelte per status waarin een actie kan worden uitgevoerd (configureerbare button)
- meerdere configureerbare 'waarschuwingskleuren' per actieve status
- achtergrondkleur voor de actieve status

#### Amsterdam mist nog
- warning/error status met eigen icoon en kleur 
- gedeelte per status waarin een actie kan worden uitgevoerd (configureerbare button)
- meerdere configureerbare 'waarschuwingskleuren' per actieve status
- achtergrondkleur voor de actieve status

+Elke hoofdstatus is collapsible, ongeacht of er substeps zijn, dus metadata kan worden ingeklapt.

#### Open Inwoner mist nog
- heeft nooit substeps want er bestaan geen zaak substatussen in Esuite / Open Zaak
- heeft geen nummers of specifieke iconen per status (afgerond/actief/toekomstige status allemaal een bolletje)
- gebruikt geen iconen voor de warning/error status (alleen bolletje met de info/success/warning/error kleur)
 
+Elke hoofdstatus is collapsible, ongeacht of er substeps zijn, dus metadata kan worden ingeklapt.

---

Huidige live variant in Open Inwoner met configureerbare achtergrondkleur van actieve status:

<img width="567" height="495" alt="Screenshot 2026-04-23 at 12 00 28" src="https://github.com/user-attachments/assets/130259c0-0848-4cf1-b7bb-a318cfba9536" />

---

### Onderzoekslinks

Onderzoek naar Process Steps Den Haag (Utrecht?):
https://gebruikersonderzoeken.nl/docs/onderzoek-bekijken/mijn-zaken/utrecht-status-component/

Onderzoek Progress List Amsterdam

Onderzoek Zaak Statussen OIP / Groningen
https://gebruikersonderzoeken.nl/docs/onderzoek-bekijken/mijn-zaken/groningen-mijn-omgeving/
